src/thor/triplegbuilder.cc

```diff
@@ -1122,12 +1133,14 @@ void TripLegBuilder::Build(
       osmchangeset = start_tile->header()->dataset_id();
     }
 
+    const bool is_first_edge = edge_itr == path_begin;
+    const bool is_last_edge = edge_itr == (path_end - 1);
+
     // have to always compute the offset in case the timezone changes along the path
     // we could cache the timezone and just add seconds when the timezone doesnt change
-    time_info = time_info.forward(trip_path.node_size() == 0
-                                      ? 0.0
-                                      : trip_path.node().rbegin()->cost().elapsed_cost().seconds(),
```

Comment:
@genadz good point! it looks like moving way into the future with this previous code. i can only imagine the code was supposed to be:

```c++
trip_path.node().rbegin()->cost().elapsed_cost().seconds() - (trip_path.node().rbegin()++)->cost().elapsed_cost().seconds()
```

Reply:
concretely this meant that the edge speed attribute was incorrect when using historical speeds

Reply:
@genadz ill quickly performance test this to make sure the timezone stuff doesnt matter and we'll get it shipped